### PR TITLE
Update DNS-SD section

### LINKED ISSUE #403: Refine DNS-SD Name Requirements and IANA Consideration
- What exactly needs to be registered?  
     - Just _wot under SRV and TXT?  
     - Do we have to register PTR?  
     - Do we have to register _directory?
- Remove editor's notes (or comment them out), at this point they are confusing
- Then update IANA registration request; for now we will tell them we are working on it (@ashimura will respond to IANA; note: actual registration can be post-CR, but fixing the text in our doc should be pre-CR)

Should also ask PLH - has anyone else done DNS-SD registrations?  (@ashimura will ask)

### PR DESCRIPTION
**Pull request by k-toumura**
Fix #403 
- Use table for describe service names.
- Clarified correspondence between service names and protocols.
- Remove Editor Notes in DNS-SD section.
- Update Assignment Notes in 11.2 Service Name Registration to describe an usage of  `_directory` subtype.
- Update diagrams (use `/.well-known/wot` instead of `/.well-known/wot-thing-description`)

Remaining issues:
- Should we use separate service names for HTTPS and CoAPS?
  - for example: `_wot-tls._tcp` and `_wot-dtls._udp` (like [CoRE-RD](https://www.rfc-editor.org/rfc/rfc9176.html#section-4.1.2))
- Should we support CoAP over TCP and CoAP over TLS?
  - for example: `_wot-coap._tcp` and `_wot-coap-tls._tcp`
- (If we add the above, do we have two (or more) implementations for them?)

***
<a href="https://pr-preview.s3.amazonaws.com/k-toumura/wot-discovery/pull/405.html" title="Last updated on Sep 5, 2022, 12:35 AM UTC (118fedb)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-discovery/405/e03c6fc...k-toumura:118fedb.html" title="Last updated on Sep 5, 2022, 12:35 AM UTC (118fedb)">Diff</a>

### CODE REVIEW

File: images/sequence-dnssd-directory.puml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 @startuml
+skin rose
 skinparam defaultFontName "Noto Sans"
 participant "Consumer" as Consumer
 participant "Directory" as Directory
 
 Consumer --> Directory : mDNS query\n"_directory._sub._wot._tcp.local PTR"
-Directory ->  Consumer : mDNS resp.\n"_directory._sub._wot._tcp.local PTR dir._wot._tcp.local"\n"dir._wot._tcp.local SRV 1 0 80 tdir.local"\n"dir._wot._tcp.local TXT td=/.well-known/wot-thing-description;type=Directory"
-Consumer -> Directory  : HTTP GET\n"http://tdir.local:80/.well-known/wot-thing-description"
+Directory ->  Consumer : mDNS resp.\n"_directory._sub._wot._tcp.local PTR dir._wot._tcp.local"\n"dir._wot._tcp.local SRV 1 0 80 tdir.local"\n"dir._wot._tcp.local TXT td=/.well-known/wot;type=Directory"
```
Comment (mmccool) [inline images/sequence-dnssd-directory.puml:8]: Expand "resp." to "response" to be clearer (pls also regen images...)
```suggestion
Directory ->  Consumer : mDNS response\n"_directory._sub._wot._tcp.local PTR dir._wot._tcp.local"\n"dir._wot._tcp.local SRV 1 0 80 tdir.local"\n"dir._wot._tcp.local TXT td=/.well-known/wot;type=Directory"
```

File: images/sequence-dnssd-thing.puml [hunk 2]
```diff
@@ -1,10 +1,11 @@
 @startuml
+skin rose
 skinparam defaultFontName "Noto Sans"
 participant "Consumer" as Consumer
 participant "Thing" as Thing
 
 Consumer --> Thing : mDNS query\n"_wot._tcp.local PTR"
-Thing ->  Consumer : mDNS resp.\n"_wot._tcp.local PTR lamp._wot._tcp.local"\n"lamp._wot._tcp.local SRV 1 0 80 lamp.local"\n"lamp._wot._tcp.local TXT td=/.well-known/wot-thing-description;type=Thing"
-Consumer -> Thing  : HTTP GET\n"http://lamp.local:80/.well-known/wot-thing-description"
+Thing ->  Consumer : mDNS resp.\n"_wot._tcp.local PTR lamp._wot._tcp.local"\n"lamp._wot._tcp.local SRV 1 0 80 lamp.local"\n"lamp._wot._tcp.local TXT td=/.well-known/wot;type=Thing"
```
Comment (mmccool) [inline images/sequence-dnssd-thing.puml:8]: as above
```suggestion
Thing ->  Consumer : mDNS response\n"_wot._tcp.local PTR lamp._wot._tcp.local"\n"lamp._wot._tcp.local SRV 1 0 80 lamp.local"\n"lamp._wot._tcp.local TXT td=/.well-known/wot;type=Thing"
```

File: index.html [hunk 3]
```diff
@@ -633,20 +641,6 @@ <h2>DNS-Based Service Discovery</h2>
...
-                <code>path</code>:
-                    The URI of the thing description on the Web Thing's web server<br/>
-                <code>td</code>:
-                    Prefix of directory service API<br/>
-                <code>tls</code>:
-                    Value of 1 if the Web Thing supports connections via HTTPS.<br/>
-            </p>
             <p>
                 <a href="#sequence-dnssd-thing"></a> and <a href="#sequence-dnssd-directory"></a> shows example sequences
                 of discovery of Thing and <a>Thing Description Directory</a> using DNS-SD and mDNS.
```
Comment (mmccool) [inline index.html:646]:
```suggestion
                supporting discovery of a Thing or a <a>Thing Description Directory</a> using DNS-SD and mDNS. 
```

File: index.html [hunk 4]
```diff
@@ -3000,7 +2990,7 @@ <h2>Service Name Registration</h2>
                 <li>Transport Protocol: TCP, UDP</li>
                 <li>Description: Service name to search a Thing or a Thing Description Directory of W3C Web of Things</li>
```
Comment (mmccool) [inline index.html:2991]:
```suggestion
                <li>Description: Service name to search for the Thing Description of a Thing or a Thing Description Directory in the W3C Web of Things</li>
```

**Review by mmccool — CHANGES_REQUESTED** [review]

**Comment by mmccool** [comment]
Looks good, I suggested a few small changes to the images to make them clearer.  Captions in figure should be improved (github would not let me make suggestions on these lines...).  In the first figure, caption could be improve by saying "Using mDNS for discovery of the Thing Description of a Thing."  In the second figure, it should say "Using mDNS for discovery of the Thing Description of a Thing Description Directory (TDD)".

I notice also that the URLs that are returned here are .well-known/wot.  While this will often be the case if both well-known and DNS introductions are supported, this is not actually required, and DNS could return another URL that does not follow the .well-known structure (this is true also for other schemes, like Direct).  I don't know if we need another statement or example to make this clearer, or should avoid using .well-known in examples to avoid confusion/assumptions on the part of the reader.

**Comment by k-toumura** [comment]
Thank you for your comment.  I've made the corrections as you indicated.

I also updated URLs to more neutral one: `/path/to/td`.

**Comment by mmccool** [comment]
Discussion:
- Problem with many service names as in CoRE-RD is many rounds of broadcast needed
- Better if one request, parameters in option
- Propose adding an additional (optional) parameter in TXT, "scheme", with values of "http", "https", "coap", and "coaps", corresponding to "scheme" part of URL.  Defaults would be "http" for TCP and "coap" for UDP

**Pull request merged**